pkg/apis/core/validation/validation.go

```diff
@@ -3873,9 +3873,37 @@ func ValidateContainerUpdates(newContainers, oldContainers []core.Container, fld
 			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("image"), ctr.Image, "must not have leading or trailing whitespace"))
 		}
 	}
+
+	// validate updated container probe
+	for i, ctr := range newContainers {
+		allErrs = append(allErrs, validateProbeUpdates(ctr.StartupProbe, oldContainers[i].StartupProbe, fldPath.Index(i).Child("startupProbe"))...)
+		allErrs = append(allErrs, validateProbeUpdates(ctr.LivenessProbe, oldContainers[i].LivenessProbe, fldPath.Index(i).Child("livenessProbe"))...)
+	}
+
 	return allErrs, false
 }
 
+func validateProbeUpdates(newProbe, oldProbe *core.Probe, fldPath *field.Path) (allErrs field.ErrorList) {
+	allErrs = field.ErrorList{}
+
+	mungedProbe := newProbe
+	if mungedProbe != nil && oldProbe != nil {
+		if oldProbe.TerminationGracePeriodSeconds != nil && *oldProbe.TerminationGracePeriodSeconds < 0 &&
+			mungedProbe.TerminationGracePeriodSeconds != nil && *mungedProbe.TerminationGracePeriodSeconds == 1 {
+			mungedProbe = mungedProbe.DeepCopy()
+			mungedProbe.TerminationGracePeriodSeconds = oldProbe.TerminationGracePeriodSeconds
+		}
+	}
+
+	// validate updated probe
+	if !reflect.DeepEqual(mungedProbe, oldProbe) {
+		probeDiff := diff.ObjectDiff(mungedProbe, oldProbe)
+		allErrs = append(allErrs, field.Forbidden(fldPath, fmt.Sprintf("probe updates may not change fields other than `terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)\n%v", probeDiff)))
```

Comment:
is changing a negative TerminationGracePeriodSeconds to 1 to fixup invalid data the only change allowed? if so, I would recommend just saying "probe updates may not change fields"